packages/react/src/components/Code/index.tsx

```diff
@@ -18,30 +18,64 @@ const Highlighter = SyntaxHighlighter as React.ComponentType<
   SyntaxHighlighterProps
 >;
 
-interface Props extends SyntaxHighlighterProps {
+type Props = {
   children: string;
   language?: 'javascript' | 'css' | 'html' | 'yaml';
   className?: string;
-  tabIndex?: number;
-}
+  scrollable?: boolean;
+} & SyntaxHighlighterProps &
+  React.HTMLAttributes<HTMLDivElement>;
 
 const Code: React.ComponentType<React.PropsWithChildren<Props>> = ({
   children,
   className,
-  tabIndex,
+  scrollable = false,
   ...props
-}) => (
-  <>
+}) => {
+  const ref = useRef<HTMLPreElement>(null);
+  const [scrollableRegion, setScrollableRegion] = useState(false);
+  // react-syntax-highlighter does not provide direct access to its dom elements
+  // via refs, but we can specify the wrapping tags to bypass this limitation
+  // see: https://github.com/react-syntax-highlighter/react-syntax-highlighter/issues/335
+  const PreWithRef = (props: React.HTMLAttributes<HTMLPreElement>) => (
+    <pre {...props} ref={ref} />
+  );
+
+  useEffect(() => {
+    let observer: ResizeObserver;
+
+    // Track the containing element because resize observer will not
+    // trigger once an element becomes scrollable
+    if (scrollable && ref.current?.parentElement) {
+      const listener: ResizeObserverCallback = () => {
+        if (!ref.current) {
+          return;
+        }
+
+        const element = ref.current;
+        setScrollableRegion(element.clientWidth < element.scrollWidth);
+      };
+      const observer = new ResizeObserver(listener);
+      observer.observe(ref.current.parentElement);
+    }
+
+    return () => observer?.disconnect();
+  }, []);
+
+  return (
     <Highlighter
       {...props}
+      PreTag={PreWithRef}
       useInlineStyles={false}
-      className={classNames('Code', className)}
-      tabIndex={tabIndex}
+      className={classNames('Code', className, {
+        'Code--scrollable': scrollable
+      })}
+      tabIndex={scrollableRegion ? 0 : undefined}
```

Comment:
This is a breaking change, right? Should we either make sure to label the commit as a breaking change or try to preserve the current functionality by continuing to allow `tabIndex` to be passed in? If we did the latter, this code could be (but maybe pulled out into a separate variable:

```suggestion
      tabIndex={tabIndex !== undefined ? tabIndex : scrollableRegion ? 0 : undefined}
```